### PR TITLE
bindings: add grove gpio header

### PR DESCRIPTION
**Pull request by joelguittet**
Add binding support for Grove header.

This will be used later for Wio Terminal board added with https://github.com/zephyrproject-rtos/zephyr/pull/54953.

This can be used later for other boards as well!

### CODE REVIEW

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please add a groove header entry to at least one in-tree board.

**Comment by joelguittet** [comment]
@henrikbrixandersen I have added initially the grove header on the board I'm currently proposing at https://github.com/zephyrproject-rtos/zephyr/pull/54953 but other maintainer required me to separate the PR.
We planned to do another PR to add the raspberrypi and grove headers on the board.

If you want to have an example here with the YAML (which I perfectly understand, I will be able to do it after the PR for the board is merged).

PS : I have no other board already in Zephyr with a Grove header.

**Comment by joelguittet** [comment]
@henrikbrixandersen this is now available.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please split this into two commits. One adding the binding, another for using it in the Wio terminal board definition.

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by joelguittet** [comment]
Thanks @henrikbrixandersen !

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]

**Comment by cfriedt** [comment]
Hi @joelguittet - looks like there is a merge conflict. Can you please rebased and fix conflicts? @henrikbrixandersen & I will approve again.

**Comment by joelguittet** [comment]
Hello @cfriedt yes normal it's due to https://github.com/zephyrproject-rtos/zephyr/pull/55189, I rebase right now! Thanks for your reviews!

**Review by cfriedt — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**
